### ISSUE TITLE
One-liner annotation causes validation error inside block 

### ISSUE THREAD

**Issue by przemekborowski** [rokucommunity/brighterscript#615]
Im trying to create annotation plugin to inject child components into parent as members (similar to some android frameworks)

The clearest syntax I discovered so far is:
```
@inject m.someChild = {} // to be changed to: m.someChild = m.top.findNode("someChild")
```
or 
```
@inject("name") m.someChild = {} // to be changed to: m.someChild = m.top.findNode("name")
```
Unfortunately such construct causes parsing error (BS1039: Expected newline or ':' at the end of a statement), even though annotation feature description says it should be possible to do it inline.

Split into two lines works, 
```
@inject 
m.someChild = {}

@inject 
m.otherChild = {}

@inject 
m.yetAnotherChild = {}

----------- vs ------------

@inject m.someChild = {}
@inject m.otherChild = {}
@inject m.yetAnotherChild = {}
```
but then code base explodes and doesn't look neat. 
Btw: if you are aware of cleaner syntax, let me know. Ideally it could look like `@inject m.someChild` but it's not an expression then

**Comment by TwitchBronBron** [comment]
Hmm. I think the main design for annotations was supposed to be to go above function/class/interface/enum statements, not above arbitrary statements inside of function bodies.

I'll try and take a look at this week to determine how much effort it would take to make these annotations work the way you want. 

An alternative would be to just use a magic function instead. 

```
inject(m.someChild)
inject(m.otherChild)
```
Then just look for all instance of those function calls with the name `inject`. However, I do prefer your annotation syntax, it helps make it clear that you're writing "meta" code there that has some type of side-effect

**Comment by przemekborowski** [comment]
Thanks, appreciate that. Plugin works like a charm. I can create PR if you think it is helpful for community. One-liner fix would be nice addition.

**Comment by TwitchBronBron** [comment]
Yeah sure, I'd welcome a PR that allows annotations to work in this way.

**Comment by przemekborowski** [comment]
@TwitchBronBron one more thing... this combination is also not working in the same context:
```
@inject @observe("itemSelected", "itemFocused")
m.someComponent = {}
```
instead:
```
@inject 
@observe("itemSelected", "itemFocused")
m.someComponent = {}
```
must be written.  I'm working on another annotation to make this:
```
@inject @observe("itemSelected", "itemFocused")
m.someComponent = {}

--------- or ---------

@inject @observe("itemSelected", "itemFocused") m.someComponent = {}
```
to be transpiled to 
```
m.someComponent = m.top.findNode("someComponent")
m.someComponent.observeField("itemSelected", "onSomeComponentItemSelected")
m.someComponent.observeField("itemFocused", "onSomeComponentItemFocused")
```
